### PR TITLE
api: support to multiple filters

### PR DESCRIPTION
**Pull request by eduardoveiga**
the operations "and" and "or" are supported in the filter

### CODE REVIEW

File: api/pkg/store/mongo/store.go [hunk 1]
```diff
@@ -25,37 +25,58 @@ func NewStore(db *mongo.Database) *Store {
...
-			if err != nil {
-				return nil, 0, err
 			}
-			queryFilter = append(queryFilter, bson.M{
-				filter.Params.Name: bson.M{
-					"$eq": operator,
-				},
-			})
 		}
+		if filter.Type == "operator" {
```
Comment (otavio) [inline api/pkg/store/mongo/store.go:59]: We have the switch case, this should be reworked and keep using it as it makes the code easier to read.

Please see if you can get this to work.

Comment (otavio) [reply api/pkg/store/mongo/store.go:59]: It seems a cleaned code, preserving your logic, would be:

```diff
diff --git a/api/pkg/store/mongo/store.go b/api/pkg/store/mongo/store.go
index c320c7c..beddd6d 100644
--- a/api/pkg/store/mongo/store.go
+++ b/api/pkg/store/mongo/store.go
@@ -28,7 +28,8 @@ func (s *Store) ListDevices(ctx context.Context, perPage, page int, filters []mo
 	var queryMatch []bson.M
 	var queryFilter []bson.M
 	for _, filter := range filters {
-		if filter.Type == "property" {
+		switch filter.Type {
+		case "property":
 			switch filter.PropertyParams.Operator {
 			case "like":
 				queryFilter = append(queryFilter, bson.M{
@@ -53,26 +54,24 @@ func (s *Store) ListDevices(ctx context.Context, perPage, page int, filters []mo
 						"$eq": operator,
 					},
 				})
-
 			}
-		}
-		if filter.Type == "operator" {
+		case "operator":
+			operator := nil
 			switch filter.OperatorParams.Name {
 			case "and":
-
-				queryMatch = append(queryMatch, bson.M{
-					"$match": bson.M{"$and": queryFilter},
-				})
-				queryFilter = []bson.M{}
+				operator = "$and"
 			case "or":
-				queryMatch = append(queryMatch, bson.M{
-					"$match": bson.M{"$or": queryFilter},
-				})
-				queryFilter = []bson.M{}
+				operator = "$or"
 			}
-		}
 
+			queryMatch = append(queryMatch, bson.M{
+				"$match": bson.M{operator: queryFilter},
+			})
+
+			queryFilter = []bson.M{}
+		}
 	}
+
 	if len(queryFilter) > 0 {
 		queryMatch = append(queryMatch, bson.M{
 			"$match": bson.M{"$or": queryFilter},
```

However, it is not clear to me why you are resetting `queryFilter` when selecting the operators. It seems odd to me.

Comment (eduardoveiga) [reply api/pkg/store/mongo/store.go]: QueryFilter is an intermediate slice to to append each filter until I got the operator
in this moment I append all values acumulated in QueryFilter to respective and/or groups in QueryMatch slice. 

To restart this routine I reset QueryFilter values.  @gustavosbarreto do you know another aproach to implement this task? This was my first

File: api/pkg/store/mongo/store.go [hunk 2]
```diff
@@ -18,6 +18,47 @@ var wrongType error = errors.New("wrong parameters type")
 
 const match string = "$match"
 
+var lookupDevices bson.M = bson.M{
+
```
Comment (otavio) [inline api/pkg/store/mongo/store.go:22]: Drop newline.

File: api/pkg/store/mongo/store.go [hunk 3]
```diff
@@ -18,6 +18,47 @@ var wrongType error = errors.New("wrong parameters type")
...
+	"$lookup": bson.M{
+		"from":         "connected_devices",
+		"localField":   "uid",
+		"foreignField": "uid",
+		"as":           "online",
+	},
+}
+
+var lookupUsers bson.M = bson.M{
+
```
Comment (otavio) [inline api/pkg/store/mongo/store.go:32]: Here as well.

File: api/pkg/store/mongo/store.go [hunk 4]
```diff
@@ -13,6 +14,10 @@ import (
 	"go.mongodb.org/mongo-driver/mongo/options"
 )
 
+var wrongType error = errors.New("wrong parameters type")
```
Comment (otavio) [inline api/pkg/store/mongo/store.go:17]: typeErr

File: api/pkg/store/mongo/store.go [hunk 5]
```diff
@@ -13,6 +14,10 @@ import (
...
+
+const match string = "$match"
```
Comment (otavio) [inline api/pkg/store/mongo/store.go:19]: The idea is good but it mixes string and consts. I'd probably avoid adding this const so the query uses a single style.

File: api/pkg/store/mongo/store.go [hunk 6]
```diff
@@ -94,29 +133,9 @@ func (s *Store) ListDevices(ctx context.Context, perPage, page int, filters []mo
...
-		},
-		{
-			"$addFields": bson.M{
-				"online":    bson.M{"$anyElementTrue": []interface{}{"$online"}},
-				"namespace": "$namespace.username",
-			},
-		},
+		lookupDevices,
+		lookupUsers,
+		onlineDevices,
```
Comment (otavio) [inline api/pkg/store/mongo/store.go:138]: This rework is very nice; it'd be good if it comes before the addition of the multiple filters so it is even easier to follow.

File: ui/src/components/device/DeviceRename.vue [hunk 7]
```diff
@@ -35,6 +35,7 @@
           You entered an invalid hostname,
           it must follow the
           <a
+            rel="noopener noreferrer"
```
Comment (gustavosbarreto) [inline ui/src/components/device/DeviceRename.vue:38]: ????

File: api/pkg/store/mongo/store.go [hunk 8]
```diff
@@ -13,6 +14,8 @@ import (
 	"go.mongodb.org/mongo-driver/mongo/options"
 )
 
+var typeErr error = errors.New("wrong parameters type")
```
Comment (gustavosbarreto) [inline api/pkg/store/mongo/store.go:17]:
```
var ErrWrongParamsType = errors.New("Wrong parameters type")
````

**Pull request merged**